macros/attribution_overview.sql

```diff
@@ -7,7 +7,76 @@
 
 {%- set __, last_processed_cv_tstamp = snowplow_utils.return_limits_from_model(ref('snowplow_attribution_campaign_attributions'),'cv_tstamp','cv_tstamp') %}
 
-with campaign_prep as (
+-- making sure we only include spend that is needed
+with spend_with_unique_keys as (
+  
+  select row_number() over() as spend_id, *
+  from {{ var('snowplow__spend_source') }}
+)
+
+, campaign_spend as (
+
+  {% if var('snowplow__spend_source') != 'not defined' %}
+    select s.campaign, s.spend
+    from spend_with_unique_keys s
+    inner join {{ ref('snowplow_attribution_campaign_attributions') }} c
+    on c.campaign = s.campaign and s.spend_tstamp < cv_tstamp 
+    and s.spend_tstamp > {{ snowplow_utils.timestamp_add('day', -90, 'cv_tstamp') }}
+    where s.campaign is not null
+    qualify row_number() over (partition by s.spend_id ) = 1
```

Comment:
It might be worth adding a comment (mainly because I didn't realise upfront) why we can't just group then join (because it has to filter using the join, not before). Nice solution though!